Framework/Nexus/src/NexusFileIO.cpp

```diff
@@ -420,7 +420,7 @@ int NexusFileIO::writeNexusProcessedData2D(const API::MatrixWorkspace_const_sptr
     start[0] = 0;
     asize[1] = dims_array[1];
     for (size_t i = 0; i < nSpect; i++) {
-      NXputslab(fileID, localworkspace->x(i).rawData().data(), start, asize);
+      NXputslab(fileID, localworkspace->x(spec[i]).rawData().data(), start, asize);
```

Comment:
Can I be slightly annoying and ask that we rename this `spec` variable? I was looking over the code and I got confused whether this contains spectrum numbers or workspace indices. I checked `SaveNexusProcessed` and it passes through [`indices`](https://github.com/mantidproject/mantid/blob/d529618f2eed7c53a7c1b90bd0afb349437c514e/Framework/DataHandling/src/SaveNexusProcessed.cpp#L305). I think renaming this variable would help the readability of the code a lot and we may as well do it while while we are changing it.